set: fix virtual reservation leak

kswiecicki opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/pmdk/5239)
<!-- Reviewable:end -->

pbalcer reviewed (commented):
lgtm, but is this only a memory leak? is there no user-visible side-effect that you could add a test for?

kswiecicki commented:
> LGTM, but is this only a memory leak? Is there no user-visible side-effect that you could add a test for?
> Why haven't Valgrind memcheck caught it?
As you have pointed out, it was not a memory leak, but a "virtual reservation leak' if I can call it that. 





<!-- Sent from Reviewable.io -->

kswiecicki commented:
___
*[src/libpmemset/pmemset.c, line 837 at r1](https://reviewable.io/reviews/pmem/pmdk/5239#-ManLkkDPzObsChm5Qte-r1-837:-MbkTam574XkFAFwJh2t:b-hwudzb) ([raw file](https://github.com/pmem/pmdk/blob/cea4a26e67cb9a315805331e4cd436c58ebf2168/src/libpmemset/pmemset.c#L837)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

I don't like using `!` with non-bools. Use explicit comparison against NULL.
Also, this looks like relaying on some internal logic that if config resv is null then we can "adjust" the reservation here. I think I understand why, but this looks odd. At the very least please add a comment.
</blockquote></details>

I've changed the comparisions and added a comment.


<!-- Sent from Reviewable.io -->
